### DISCUSSION TITLE
How to use internal validation with QEditor? Should I be wrapping it in a QField?

### DISCUSSION THREAD

**Discussion by BenJackGill** [quasarframework/quasar#14085] [General - Components / Directives / etc]
I want to use internal validation with the QEditor.

Specifically, when the WYSIWYG editor is empty I want an error to show. For example, this is how internal validation looks on an empty QInput, and I want the same thing on my QEditor:
![Screen Shot 2022-07-29 at 9 52 37 am](https://user-images.githubusercontent.com/2043257/181655805-19b9a978-047d-46d4-927f-a17aa923a9ec.png)

Currently I am attempting to achieve this by wrapping QEditor in a QField.

This is what my code looks like:

```Vue
<q-field
  v-model="editorContent"
  :rules="[(val: string) => !!val || 'Please include ticket details.']"
>
  <q-editor
    ref="editorElement"
    v-model="editorContent"
    placeholder="Type your ticket here..."
    min-height="15rem"
    style="width: 100%"
    :toolbar="[
      ['bold', 'italic', 'strike', 'underline'],
      ['unordered', 'ordered'],
      ['link'],
      ['undo', 'redo'],
    ]"
    @paste="handleEditorPaste"
  />
</q-field>
```

And it almost works.

When the page first loads and I click into the editor and immediately click out of the editor it shows the validation error correctly.

But if I type something, delete that text, and then click out of the editor, no validation error shows. Doing this action should be triggering the same validation error as above because there is no text in the editor, but it does not work.

Any ideas on how to get this working?

P.S. I tried to re-create this in a codepen but it wouldn't handle the `:rules="[(val: string) => !!val || 'Please include ticket details.']"` correctly. Just showed a blank page when that was included.

**Comment by BenJackGill** [comment]
Ok I figured out the first problem.

When deleting the content of the editor it was leaving behind an invisible `<br>` element.

So I removed the `<br>` using the `@update:model-value` event like so: 

```Vue
      <q-field
        ref="editorFieldElement"
        v-model="editorContent"
        :rules="[(val: string) => !!val || 'Please include ticket details.']"
      >
        <q-editor
          ref="editorElement"
          v-model="editorContent"
          placeholder="Type your ticket here..."
          min-height="15rem"
          style="width: 100%"
          :toolbar="[
            ['bold', 'italic', 'strike', 'underline'],
            ['unordered', 'ordered'],
            ['link'],
            ['undo', 'redo'],
          ]"
          @paste="handleEditorPaste"
          @update:model-value="handleEditorChange" // <-- Added this here
        />
      </q-field>
```

```TypeScript
const handleEditorChange = (value: string) => {
  if (value === '<br>') {
    editorContent.value = '';
  }
};
```

But I have another problem with this solution...

When there is an input validation error the exclamation mark appears on top right hand corner, and causing the editor to be pushed in at the side. So I'm wondering if I can move that exclamation mark _inside_ the editor to make it more visually appealing.

Here is a screenshot to explain:

![Screen Shot 2022-08-29 at 9 31 59 am](https://user-images.githubusercontent.com/2043257/187099374-65061834-c285-4114-a5e6-e3ee58095c15.png)

**Reply by huynt0911** [reply]
Hi @BenJackGill, I am facing the same issue. Have you figured out any workaround for the input validation error in the top right corner?

**Comment by BenJackGill** [comment]
Sorry I switched away to PrimeVue, not sure about Quasar stuff anymore.

**Comment by jop-io** [comment]
My workaround for the error icon is basically to just hide it. The editor will scream in red color any way, so it is hard for a user to miss where to look :-)

If you still want the error icon to show inside the editor, I guess you could just use position absolute or something similar.

My workaround looks like this:

```html
<!--- class .editor-wrapper added to q-field -->
<q-field
  class="editor-wrapper"
  borderless
  ref="editorFieldElement"
  v-model="editorContent"
  :rules="[(val: string) => !!val || 'Please include ticket details.']"
>
  <q-editor
    ref="editorElement"
    v-model="editorContent"
    placeholder="Type your ticket here..."
    min-height="15rem"
    style="width: 100%"
    :toolbar="[
      ['bold', 'italic', 'strike', 'underline'],
      ['unordered', 'ordered'],
      ['link'],
      ['undo', 'redo'],
    ]"
    @paste="handleEditorPaste"
    @update:model-value="handleEditorChange"
  />
</q-field>

<style>
.editor-wrapper .q-field__append {
  display: none;
  /* you should be able to use position absolute (or similar) here, instead of just hiding the icon */
}
</style>
```
